assets/JS/index.js

```diff
@@ -0,0 +1,117 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+//arrays used to generate PW
+const alphaLower = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];
+const alphaUpper = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"];
+const numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+const special = ["!", '"', "#", "%", "&", "'", "(", ")", "*", "+", "," ,"-",".","/",":",";","<","=",">","?","@","[","]","^","_","`","{","|","}","~"];
+
+// main function to generate the random password
+const generatePassword = () => {
+  
+
+// get the password length
+const passwordLength = getPasswordLength();
+
+if (passwordLength) {
+// get the password criteria
+  const passwordCriteria = getPasswordCriteria();
+//test if user input is a number and within the correct parameters
+    if (Number.isNaN(passwordLength)) {
+      alert("Incorrect value, please choose a valid number.");
+      return false;
+
+      } else {
+        // create random password
+        const password = createRandomPassword(passwordLength, passwordCriteria);
+        return password;
+      };
+    }
+  }
+
+// get password length
+const getPasswordLength = () => {
+  //ask user how long password should be
+  const passLength = prompt("Please choose password length between 8 and 128.", "18");
+
+  const determinePassLength = parseInt (passLength, 10);
+// if you input something other than what is specified you will run into an error
+  if (determinePassLength >= 8 && determinePassLength <= 128){
+
```

Comment:
If the `if` block is empty, that means we probably don't need it in the first place. Why don't we just check for the negative of this condition `if (determinePassLength < 8 || determinePassLength > 128)` and then throw the alert then